services/ui-src/src/components/modals/AddNewWorkPlanModal.tsx

```diff
@@ -0,0 +1,194 @@
+import { useContext, useState } from "react";
+// components
+import { Form, Modal, ReportContext } from "components";
+import newWpFormJson from "forms/addEditWpReport/newWpReport.json";
+import { AnyObject, FormJson, ReportStatus } from "types";
+import { injectFormWithReportPeriodYears, useStore } from "utils";
+import { States } from "../../constants";
+
+export const AddNewWorkPlanModal = ({
+  activeState,
+  reportType,
+  modalDisclosure,
+}: Props) => {
+  const { createReport, fetchReportsByState } = useContext(ReportContext);
+  const { full_name } = useStore().user ?? {};
+  const [submitting, setSubmitting] = useState<boolean>(false);
+
+  /**
+   * @function: Generate report year choices
+   * Populate previous, current, and next year for the choice list,
+   * except if it is the first year of the work plan feature
+   */
+  const generateReportYearChoices = () => {
+    const WP_LAUNCH_YEAR = 2024;
+    const currentYear = new Date(Date.now()).getFullYear();
+    let yearList = [];
+
+    if (currentYear === WP_LAUNCH_YEAR) {
+      yearList = [
+        {
+          label: `${currentYear}`,
+          id: `reportYear-${currentYear}`,
+          value: `${currentYear}`,
+        },
+        {
+          label: `${currentYear + 1}`,
+          id: `reportYear-${currentYear + 1}`,
+          value: `${currentYear + 1}`,
+        },
+      ];
+    } else {
+      yearList = [
+        {
+          label: `${currentYear - 1}`,
+          id: `reportYear-${currentYear - 1}`,
+          value: `${currentYear - 1}`,
+        },
+        {
+          label: `${currentYear}`,
+          id: `reportYear-${currentYear}`,
+          value: `${currentYear}`,
+        },
+        {
+          label: `${currentYear + 1}`,
+          id: `reportYear-${currentYear + 1}`,
+          value: `${currentYear + 1}`,
+        },
+      ];
+    }
+
+    return yearList;
+  };
+
+  const modalFormJson = injectFormWithReportPeriodYears(
+    newWpFormJson,
+    generateReportYearChoices()
+  );
+  const form: FormJson = modalFormJson;
+
+  /**
+   * @function: Prepare WP payload
+   * @param wpReset: (optional) determine whether the user would like to continue a Work Plan for next period,
+   * but clear / reset any existing data from the previous period
+   */
+  const prepareWpPayload = (formData: any, wpReset?: boolean) => {
+    const submissionName = "Work Plan";
+    const formattedReportYear = Number(formData.reportPeriodYear[0].value);
+    const formattedReportPeriod =
+      formData.reportPeriod[0].key === "reportPeriod-1" ? 1 : 2;
+    // static entities
+    const targetPopulations = [
```

Comment:
I think this is also in `AddEditReportModal.tsx` also. Since it's in two files maybe it can be put in a modal utility file and exported.

Reply:
good idea! I also noticed some overlap between this new modal and the AddEdit modal. wondering how much could be reused 🤔

Reply:
For now, I think it's best to pull the `targetPopulations` out into a const that can be shared. I started tinkering with extracting the entire `prepareWpPayload` function into a util, but there are a few differences between the two modals and the information they pass to that function, so I'm going to vote on punting on that refactor and capturing it into a ticket. (And that ticket is [here](https://jiraent.cms.gov/browse/CMDCT-3473)!)

Reply:
Agreed, this looks good.